sklearn/manifold/locally_linear.py

```diff
@@ -68,10 +68,9 @@ def barycenter_kneighbors_graph(X, n_neighbors, reg=1e-3, n_jobs=1):
 
     Parameters
     ----------
-    X : {array-like, sparse matrix, BallTree, KDTree, NearestNeighbors}
+    X : array-like
```

Comment:
I don't know this code well at all but I found at least one instance where a NearestNeighbors object can be passed as the first argument [here](https://github.com/lmcinnes/scikit-learn/blob/07aa1c443e609b89e19140950dce0d6677093d6b/sklearn/manifold/locally_linear.py#L303). What made you reach the conclusion it was not supported?